src/main/scala/devices/debug/Debug.scala

```diff
@@ -722,6 +721,263 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
       }
     }
 
+    //--------------------------------------------------------------
+    // System Bus Access
+    //--------------------------------------------------------------
+    // --- SBCS Registers ---
+
+    sb2tlOpt match {                   
+      case Some(sb2tl) => {
+       
+        /* Flags for setting error flag registers, needed because the number of SBDATA and
+         * SBADDRESS registers depends on system bus widths
+         */
+        val anyAddressWrEn = Wire(init = false.B)
+        val anyDataRdEn    = Wire(init = false.B)
+        val anyDataWrEn    = Wire(init = false.B)
+
+        // --- SBCS Status Register ---
+        val SBCSFieldsReg      = Reg(new SBCSFields())
+        val SBCSFieldsRegReset = Wire(init = (new SBCSFields()).fromBits(0.U))
+        val SBCSRdData         = Wire(init = new SBCSFields().fromBits(0.U)) 
+        val SBCSWrDataVal      = Wire(init = 0.U(32.W))
+        val SBCSWrData         = Wire(init = new SBCSFields().fromBits(SBCSWrDataVal))
+        val SBCSRdEn, SBCSWrEn = Wire(init = false.B)
+
+
+        // --- System Bus Registers ---
+        val SBADDRESS0FieldsReg = Reg(new SBADDRESS0Fields())
+        val SBADDRESS1FieldsReg = Reg(new SBADDRESS1Fields())
+        val SBADDRESS2FieldsReg = Reg(new SBADDRESS2Fields())
+        val SBADDRESS3FieldsReg = Reg(new SBADDRESS3Fields())
+        val SBADDRESS0RdData    = Wire(init = new SBADDRESS0Fields().fromBits(0.U))
+        val SBADDRESS1RdData    = Wire(init = new SBADDRESS1Fields().fromBits(0.U))
+        val SBADDRESS2RdData    = Wire(init = new SBADDRESS2Fields().fromBits(0.U))
+        val SBADDRESS3RdData    = Wire(init = new SBADDRESS3Fields().fromBits(0.U))
+
+        val SBADDRESS0WrData, SBADDRESS1WrData, SBADDRESS2WrData, SBADDRESS3WrData = Wire(init = 0.U(32.W))
+        val SBADDRESS0RdEn,   SBADDRESS1RdEn,   SBADDRESS2RdEn,   SBADDRESS3RdEn   = Wire(init = false.B)  
+        val SBADDRESS0WrEn,   SBADDRESS1WrEn,   SBADDRESS2WrEn,   SBADDRESS3WrEn   = Wire(init = false.B)  
+
+        val autoIncrementedAddr = Wire(init = 0.U(128.W))
+        autoIncrementedAddr := sb2tl.module.io.addrIn + (1.U << SBCSFieldsReg.sbaccess)
+
+        // ADDR0 Register is required
+        when (~io.dmactive) {
+          SBADDRESS0FieldsReg := new SBADDRESS0Fields().fromBits(0.U(32.W))
+        }.otherwise {
+          SBADDRESS0FieldsReg := Mux(SBADDRESS0WrEn && !SBCSFieldsReg.sberror && !SBCSFieldsReg.sbbusy, new SBADDRESS0Fields().fromBits(SBADDRESS0WrData),
+                                 Mux(sb2tl.module.io.dataOutValid && SBCSFieldsReg.sbautoincrement,     new SBADDRESS0Fields().fromBits(autoIncrementedAddr(31,0)),SBADDRESS0FieldsReg))
+        }
+
+        SBADDRESS0RdData := SBADDRESS0FieldsReg
+
+        dmiNode.regmap(
```

Comment:
I didn't think you could do this, as it would override/get overriden by the other dmiNode.regmap() calls. I could be wrong (and if I am my mind is blown because a lot of things I have worked on would have been way easier), but I *think* you need to do something like:

```
val  (sbaddr0fields, sbaddr1fields, ...)= sb2tlOpt.map { .... 
...
val addr0fields = Seq(RegField(...))
...
(addr0fields, addr1fields, ...)
}.getOrElse{(Nil, Nil ...)}
```

Then later in your regmap call:

```
dmiNode.regmap(
...
(DMI_SBADDRESS0 << 2)  -> sbaddr0fields,
(DMI_SBADDRESS1 << 2)  -> sbaddr1fields,
...
)
```

Reply:
Am I correct @terpstra ? We can't just call `dmiNode.regmap()` several times, right?

Reply:
Yeah, Wes confirmed I can't make multiple regmap() calls. I'll work on a change